### PR TITLE
fix player argument type

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Fixes issues with the scan command by fixing the player argument type

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/commands/ActionMenuState.java [hunk 1]
```diff
@@ -116,6 +116,21 @@ public EntityPlayerMP getPlayerForArgument(@NotNull final String argumentName)
         return null;
     }
 
+    public EntityPlayerMP getOnlinePlayerForArgument(@NotNull final String argumentName)
+    {
+        for (final ActionArgument actionArgument : getAllArgumentsList())
+        {
+            if (argumentName.equals(actionArgument.getName()))
+            {
+                if (ActionArgumentType.ONLINE_PLAYER == actionArgument.getType())
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/commands/ActionMenuState.java]: MAJOR Merge this if statement with the enclosing one. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 54 issues
* ![MAJOR][MAJOR] 53 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/aed47aea61cd8350cb6bb9ec14637159d56d75fa/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/aed47aea61cd8350cb6bb9ec14637159d56d75fa/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/aed47aea61cd8350cb6bb9ec14637159d56d75fa/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/aed47aea61cd8350cb6bb9ec14637159d56d75fa/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/aed47aea61cd8350cb6bb9ec14637159d56d75fa/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/aed47aea61cd8350cb6bb9ec14637159d56d75fa/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/aed47aea61cd8350cb6bb9ec14637159d56d75fa/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/aed47aea61cd8350cb6bb9ec14637159d56d75fa/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [locations#](https://github.com/Minecolonies/minecolonies/blob/aed47aea61cd8350cb6bb9ec14637159d56d75fa/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations#): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [management#](https://github.com/Minecolonies/minecolonies/blob/aed47aea61cd8350cb6bb9ec14637159d56d75fa/src/main/java/com/minecolonies/coremod/colony/requestsystem/management#): Add a 'package-info.java' file to document the 'management' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by mkienenb — DISMISSED** [review]
Yes, online player makes more sense.  Can you check for any other commands with online-player type definitions and make sure they also call the getOnlinePlayerFromArgument method?

**Review by marchermans — DISMISSED** [review]

**Review by mkienenb — CHANGES_REQUESTED** [review]
Actually, in retrospect, it would be better to change getPlayerForArgument() to support both online and "any" player types rather than creating getOnlinePlayerForArgument()

**Comment by Raycoms** [comment]
Done

File: src/main/java/com/minecolonies/coremod/commands/ActionMenuState.java [hunk 2]
```diff
@@ -107,7 +107,7 @@ public EntityPlayerMP getPlayerForArgument(@NotNull final String argumentName)
         {
             if (argumentName.equals(actionArgument.getName()))
             {
-                if (ActionArgumentType.PLAYER == actionArgument.getType())
+                if (ActionArgumentType.ONLINE_PLAYER == actionArgument.getType() || ActionArgumentType.PLAYER == actionArgument.getType())
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/commands/ActionMenuState.java:110]: MAJOR Merge this if statement with the enclosing one. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 54 issues
* ![MAJOR][MAJOR] 53 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/1394dd717a2eec93edcd900ab29e97c75cb31930/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/1394dd717a2eec93edcd900ab29e97c75cb31930/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/1394dd717a2eec93edcd900ab29e97c75cb31930/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/1394dd717a2eec93edcd900ab29e97c75cb31930/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [BuildingBuilder.java#L121](https://github.com/Minecolonies/minecolonies/blob/1394dd717a2eec93edcd900ab29e97c75cb31930/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java#L121): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/1394dd717a2eec93edcd900ab29e97c75cb31930/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/1394dd717a2eec93edcd900ab29e97c75cb31930/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/1394dd717a2eec93edcd900ab29e97c75cb31930/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/1394dd717a2eec93edcd900ab29e97c75cb31930/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [locations#](https://github.com/Minecolonies/minecolonies/blob/1394dd717a2eec93edcd900ab29e97c75cb31930/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations#): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**